### DISCUSSION TITLE
[CONTRIBUTION] Speech Dataset Generator

### DISCUSSION THREAD

**Discussion by davidmartinrius** [rsxdalv/TTS-WebUI#277] [Show and tell]
Hi everyone! 

I have just published this project on GitHub: https://github.com/davidmartinrius/speech-dataset-generator/

Now you can create datasets automatically with any audio or lists of audios.

I hope you can find it useful. 

Here are the key functionalities of the project:

1. **Dataset Generation:** The project allows for the creation of datasets with Mean Opinion Score (MOS).

2. **Silence Removal:** It includes a feature to remove silences from audio files, enhancing the overall quality.

3. **Sound Quality Improvement:** The project focuses on improving the quality of the audio.

4. **Audio Segmentation:** It can segment audio files within specified second ranges.

5. **Transcription:** The project transcribes the segmented audio, providing a textual representation.

6. **Gender Identification:** It identifies the gender of each speaker in the audio.

7. **Pyannote Embeddings:** Utilizes pyannote embeddings for speaker detection across multiple audio files.

8. **Automatic Speaker Naming:** Automatically assigns names to speakers detected in multiple audios.

9. **Multiple Speaker Detection:** Capable of detecting multiple speakers within each audio file.

Feel free to explore the project at https://github.com/davidmartinrius/speech-dataset-generator

**Comment by rsxdalv** [comment]
Great work!

**Reply by davidmartinrius** [reply]
Thank you @rsxdalv I hope it is useful for you :smile:

**Reply by rsxdalv** [reply]
Also I'm very impressed that you have such a collection of MIT licensed tools, there are many times when there's a GPL license in a project that makes it impossible for me to incorporate.

**Reply by davidmartinrius** [reply]
Oh well, whisperx is under BSD-4-Clause license, but actually I do not know if it allows you to incorporate the code... does it?

**Reply by rsxdalv** [reply]
Thanks for pointing that out. It's not immediately clear if that would force my project to change to BSD4 or it just requires me to have additional information and License for that aspect. My immediate strategy is usually to skip those projects/features initially. 
By the way, the links for whisperx and whisper-fast are broken.
